src/autumnaton.ts

```diff
@@ -0,0 +1,189 @@
+  const autumnItem = AutumnAton.getUniques(location)?.item;
+  const seasonalItemDrops = seasonalOverride ?? AutumnAton.seasonalItems();
+  if (autumnItem) {
+    return (
+      (seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0) +
+      (autumnMeltables.includes(autumnItem)
+        ? // If we already have the meltable, then we get a random item, else value at 0
+          availableAmount(autumnItem) > 0
+          ? avgValueOfRandomAutumnItem
+          : 0
+        : garboValue(autumnItem, true))
+    );
+  } else {
+    // If we're in a location without any uniques, we still get cowcatcher items
+    return seasonalItemDrops > 1 ? avgValueOfRandomAutumnItem : 0;
+  }
+}
+
+function expectedRemainingExpeditions(legOverride?: number): number {
+  const availableAutumnatonTurns = estimatedTurns() - AutumnAton.turnsLeft();
+  let expeditionTurnSum = 0;
+  let quests = get("_autumnatonQuests", 0);
+  while (expeditionTurnSum < availableAutumnatonTurns) {
+    expeditionTurnSum +=
+      11 *
+      Math.max(
+        1,
+        quests -
+          (legOverride ?? AutumnAton.currentUpgrades().filter((u) => u.includes("leg")).length)
+      );
+    quests++;
+  }
+
+  return quests - get("_autumnatonQuests", 0);
+}
+
+const profitRelevantUpgrades = [
+  "leftarm1",
+  "leftleg1",
+  "rightarm1",
+  "rightleg1",
+  "cowcatcher",
+  "periscope",
+  "radardish",
+] as const;
+
+export function mostValuableUpgrade(fullLocations: Location[]): Location[] {
+  // This function shouldn't be getting called if we don't have an expedition left
+  if (expectedRemainingExpeditions() < 1) {
+    return fullLocations;
+  }
+  const currentUpgrades = AutumnAton.currentUpgrades();
+  const acquirableUpgrades = profitRelevantUpgrades.filter(
+    (upgrade) => !currentUpgrades.includes(upgrade)
+  );
+
+  if (acquirableUpgrades.length === 0) {
+    return fullLocations;
+  }
+
+  const currentBestLocation = maxBy(fullLocations, (loc: Location) => averageAutumnatonValue(loc));
+  const currentExpectedProfit =
+    averageAutumnatonValue(currentBestLocation) * expectedRemainingExpeditions();
+
+  const upgradeValuations = acquirableUpgrades.map((upgrade) => {
+    const upgradeLocations = fullLocations.filter(
+      (location) => AutumnAton.getUniques(location)?.upgrade === upgrade
+    );
+    const bestLocContainingUpg = maxBy(upgradeLocations, (loc: Location) =>
+      averageAutumnatonValue(loc)
+    );
+
+    if (upgrade === ("periscope" || "radardish")) {
+      const bestLocWithInstalledUpg = maxBy(fullLocations, (loc: Location) =>
+        averageAutumnatonValue(loc, AutumnAton.visualAcuity() + 1)
+      );
+      const extraExpectedProfit =
+        averageAutumnatonValue(bestLocContainingUpg) +
+        averageAutumnatonValue(bestLocWithInstalledUpg) *
+          Math.max(0, expectedRemainingExpeditions() - 1);
+
+      return { upgrade, profit: extraExpectedProfit };
+    }
+    if (upgrade === ("rightleg1" || "leftleg1")) {
```

Comment:
```suggestion
    if (upgrade.includes("leg1")) {
```

We can also do the other way. Many array methods also are string methods, because a string can be thought of as an array of unit-length strings